app/models/user.rb

```diff
@@ -158,6 +159,51 @@ def self.authenticate(options)
     user
   end
 
+  aasm :column => :status, :no_direct_assignment => true do
+    state :pending, :initial => true
+    state :active
+    state :confirmed
+    state :suspended
+    state :deleted
+
+    # A normal account is active
+    event :activate do
+      transitions :from => :pending, :to => :active
+    end
+
+    # Used in test suite, not something that we would normally need to do.
+    event :deactivate do
+      transitions :from => :active, :to => :pending
+    end
```

Comment:
Well if it was only used for test cases then we should just remove those test cases as by definition we would be testing functionality that wasn't used in real life - that's better than polluting the source with comments.

As I said in my review though currently there is a production use case, it's just that this PR currently removes the ability to use it. There's probably no need for it though, if I was able to move people from suspended/confirmed to active though.

Reply:
It's only used in the factory, not in test cases. In the tests we want to default to having users in the "active" state for the vast, vast majority of test cases, but then there's no way to get a user back into the "pending" state for those tests that need it, because direct attribute assignment is disabled.

I think rather than a guard I might change it to being wrapped in the check, so that if you e.g. make a diagram using aasm-diagram it doesn't show up.

I'm open to better ideas - I thought that there might be a way in FactoryBot to set the attribute without aasm complaining but I couldn't find out how. See also https://github.com/aasm/aasm/issues/523